src/Val.cc

```diff
@@ -1420,6 +1421,115 @@ static void find_nested_record_types(const TypePtr& t, std::set<RecordType*>* fo
     }
 }
 
+// Support class for returning multiple values from a table[pattern]
+// when indexed with a string.
+class detail::TablePatternMatcher {
+public:
+    TablePatternMatcher(const TableVal* _tbl, TypePtr _yield) : tbl(_tbl) {
+        vtype = make_intrusive<VectorType>(std::move(_yield));
+    }
+
+    void Insert(ValPtr pat, ValPtr yield) { Clear(); }
+    void Remove(ValPtr pat) { Clear(); }
```

Comment:
Are these just supposed to clear the state on insert/assign? Then the method names seem misleading, I'd just skip the methods and call clear directly.

Reply:
FWIW, it has that form so that the "how" of what `TablePatternMatcher` responds to inserts and removals remains isolated to its internals. (Originally it did more than just `Clear()`, so the isolation was more material.)

Reply:
I had almost removed them, will now do.